apiary.apib

```diff
@@ -412,6 +520,151 @@ A Shop resource.
+                        {
+                                "number": "12345667",
+                                "credential": {
+                                        "provider": "visa"
+                                        "login": "12345667",
+                                        "password": "my_password"
+                                }
+                        },
+                        {
+                                "number": "912304413",
+                                "credential": {
+                                        "provider": "amex"
+                                        "login": "myshop01",
+                                        "password": "my_password"
+                                }
+                        }
+                ]
+            }
++ Response 200 (application/json)
+
+        {
+            "id": 1,
+            "name": "My Shop",
+            "stablishments": [
+                        {
+                                "id": 1
+                                "provider": "visa"
+                                "number": "12345667"
+                        },
+                        {
+                                "id": 2
+                                "provider": "amex"
+                                "number": "912304413"
+                        }
+                ]
+        }
+
++ Response 400 (application/json)
+
+        { "errors": [{"stablishments[0].credential.login": "credential already exists"}] }
+
++ Response 403 (application/json)
+
+        { "errors": ["Action forbidden"] }
+
++ Request (application/json)
+
+    + Header
+
+            Accept: application/json
+            Authorization: un_authorized_token
+
++ Response 401 (application/json)
+
+        {   error: "Not authorized"   }
+
+
+### Update a Shop [PATCH]
+
+Updates a shop. May change name or stablishments associated
+
++ Request (application/json)
+
+    + Header
+
+            Accept: application/json
+            Authorization: eyJ0eXAiOiJKV1QiLCJhbiAoOiJIUzI1NiJ9.eyJvX2F1dGhfYXBwbGljYXRpb25faWQiOjF9.h7ZNPpbd_Bu8yL27fmCAI_wgyZJI44b5eRRxwWUcwaQ
+
+    + Body
+
+            "shop": {
+                "name": "My Shop Renamed"
+            }
+
++ Response 200 (application/json)
+
+        {
+            "id": 1,
+            "name": "My Shop Renamed",
+            "stablishments": [
+                {
+                    "id": 1
+                    "provider": "visa"
+                    "number": "12345667"
+                },
+                {
+                    "id": 2
+                    "provider": "amex"
+                    "number": "912304413"
+                }
+            ]
+        }
+
++ Request (application/json)
+
+    + Header
+
+            Accept: application/json
+            Authorization: eyJ0eXAiOiJKV1QiLCJhbiAoOiJIUzI1NiJ9.eyJvX2F1dGhfYXBwbGljYXRpb25faWQiOjF9.h7ZNPpbd_Bu8yL27fmCAI_wgyZJI44b5eRRxwWUcwaQ
+
+    + Body
+
+            "stablishment_numbers": ["12345667"]
```

Comment:
instead of sending this list, did you consider sending the same Shop object, and inside api detect if any of those stablishments do not belong and try adding them? kind of like telling the api hey! this is how this shop looks like now.

Reply:
because api users could wonder if this means add or replace all for this list

Reply:
Hmm this is the standard way the PATCH verb works, replaces all named fields with the values provided. So if for example you have:
```ruby
shop: {
  id: 1,
  name: 'A Name'
  stablishments: [ {id: 1, number: '1234567'} ]
}
```
and you do: PATCH /shops/1 {shop: {stablishment_numbers: ['99123456'] } }
You'll have:
shop: {
  id: 1,
  name: 'A Name'
  stablishments: [ {id: 2, number: '99123456'} ]
}
```
given a stablishment with id 2 exists with number 99123456

Reply:
The way you suggest is the standard for PUT, which is also correct, it's just a matter of taste, we could provide both, I just felt patch made more sense here

Reply:
Ok yeah i was expecting here a PUT request, the PATCH method seems ok in theory, you should then improve the example, to clarify that sending a new stablishment_number will NOT append that number to the current list of stablishments. This action by anyone reading the docs should be cristal clear.

Reply:
in summary, ok go for patch, add some examples that include lists of numbers not just a 1 element list